### PR TITLE
Fix compile/build/tests to use released 6.4 jars.

### PR DESCRIPTION
**Pull request by rmuir**
Build no longer invokes lucene's ant build.

Upgrade lucene 6.3 -> 6.4, mostly just involves DoubleValuesSource.

Compile now works and tests run! But two of the tests fail.

### CODE REVIEW

**Comment by rmuir** [comment]
Both of the test failures look like this:

```
Server error:
java.lang.NullPointerException: Cannot read field "writer" because "this.writableShard" is null
	at org.apache.lucene.server.IndexState.getWritableShard(IndexState.java:1040)
	at org.apache.lucene.server.handlers.BulkAddDocumentHandler.handleStreamed(BulkAddDocumentHandler.java:427)
	at org.apache.lucene.server.Server$ServerHandler._handle(Server.java:391)
	at org.apache.lucene.server.Server$ServerHandler.handle(Server.java:340)
```

File: build.py [hunk 1]
```diff
@@ -13,44 +13,58 @@
...
+  ('org.apache.lucene', 'lucene-misc', LUCENE_VERSION),
+  ('org.apache.lucene', 'lucene-queries', LUCENE_VERSION),
+  ('org.apache.lucene', 'lucene-queryparser', LUCENE_VERSION),
+  ('org.apache.lucene', 'lucene-suggest', LUCENE_VERSION),
+  ('org.apache.lucene', 'lucene-expressions', LUCENE_VERSION),
+    ('org.antlr', 'antlr4-runtime', '4.5.1-1'),
+    ('org.ow2.asm', 'asm', '5.1'),
+    ('org.ow2.asm', 'asm-commons', '5.1'),
+  ('org.apache.lucene', 'lucene-replicator', LUCENE_VERSION),
+    ('org.apache.httpcomponents', 'httpclient', '4.4.1'),
```
Comment (mikemccand) [inline build.py:44]: Whoa, cool -- you did transitive closure, manually, here, so we do not even look at `ant` build files anymore from Lucene?

**Review by mikemccand — APPROVED** [review]
Amazing, only two tests failing!  Thanks @rmuir!

**Comment by mikemccand** [comment]
> java.lang.NullPointerException: Cannot read field "writer" because "this.writableShard" is null

This is cool -- is this a recent JVM change, that it tells you which field is not accessible because this other field is `null`?

**Pull request merged**